### ISSUE TITLE
incorrect default interface picked for accessing tailscale control

### ISSUE THREAD

**Issue by frankli0324** [tailscale/tailscale#3620]
### What is the issue?

tailscale is able to connect to tailsacle control plane over dhcp networks (Direct Ethernet / WLAN), but is unable to make any requests over pppoe networks. it says "network unreachable" for literally any request.

```
bootstrapDNS("derp4d.tailscale.com", "[this_ip]") for "log.tailscale.io" error: Get "https://derp4d.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp [this_ip]:443: connect: network is unreachable
```

### Steps to reproduce

connect to a PPPoE network on Mac
sudo tailscaled

### OS

macOS

### OS version

12.0.1

### Tailscale version

1.18.2

**Comment by bradfitz** [comment]
We're going to need more log context to see what's going on.

**Comment by frankli0324** [comment]
Sorry for the late reply, just woke up.

After a little poking around, I found that tailscale picked the wrong interface for the internet.
I had zerotier(another VPN) installed on my machine, and it's interface was picked

this is from tailscale startup

```
logtail started
Program starting: v1.18.2-tb04815c9c, Go 1.17.5: []string{"/usr/local/opt/tailscale/bin/tailscaled"}
LogID: [redacted]
logpolicy: using system state directory "/Library/Tailscale"
wgengine.NewUserspaceEngine(tun "utun") ...
dns: using dns.noopManager
link state: interfaces.State{defaultRoute=feth1724 ifs={feth1724:[10.[redacted]/24 fc3e:[redacted]::1/40] ppp0:[[redacted]/24] vnic0:[10.211.55.2/24] vnic1:[10.37.129.2/24]} v4=true v6=true}
```

and this comes from `sudo tailscaled debug --ifconfig`

```
❯ sudo /usr/local/opt/tailscale/bin/tailscaled debug --ifconfig
{
    "InterfaceIPs": {
        "VHC128": null,
        "XHC0": null,
        "XHC1": null,
        "XHC20": null,
        "ap1": null,
        "awdl0": null,
        "bridge0": null,
        "en0": null,
        "en1": null,
        "en2": null,
        "en3": null,
        "en4": null,
        "en5": [
            "fe80::aede:48ff:fe00:1122/64"
        ],
        "en8": [ // this is my ethernet interface for PPPoE dialing
            "169.254.133.213/16",
            "fe80::32:8928:f801:d488/64"
        ],
        "feth1724": [ // this is my zerotier interface
            "10.[redacted]/24",
            "fc3e:[redacted]::1/40",
            "fe80::[redacted]/64"
        ],
        "feth6724": null,
        "gif0": null,
        "llw0": null,
        "lo0": [
            "127.0.0.1/8",
            "::1/128",
            "fe80::1/64"
        ],
        "ppp0": [ // I expected tailscale to pick this
            "[redacted]/24",
            "fe80::167d:daff:fe6a:4478/64"
        ],
        "stf0": null,
        "utun0": [
            "fe80::3617:5f87:19e2:ac10/64"
        ],
        "utun1": [
            "fe80::3ca7:a6e:cd01:cac9/64"
        ],
        "utun2": [
            "fe80::ce81:b1c:bd2c:69e/64"
        ],
        "vnic0": [
            "10.211.55.2/24"
        ],
        "vnic1": [
            "10.37.129.2/24"
        ]
    },
    "Interface": {
        "VHC128": {
            "Index": 7,
            "MTU": 0,
            "Name": "VHC128",
            "HardwareAddr": null,
            "Flags": 0,
            "AltAddrs": null
        },
        "XHC0": {
            "Index": 4,
            "MTU": 0,
            "Name": "XHC0",
            "HardwareAddr": null,
            "Flags": 0,
            "AltAddrs": null
        },
        "XHC1": {
            "Index": 5,
            "MTU": 0,
            "Name": "XHC1",
            "HardwareAddr": null,
            "Flags": 0,
            "AltAddrs": null
        },
        "XHC20": {
            "Index": 6,
            "MTU": 0,
            "Name": "XHC20",
            "HardwareAddr": null,
            "Flags": 0,
            "AltAddrs": null
        },
        "ap1": {
            "Index": 9,
            "MTU": 1500,
            "Name": "ap1",
            "HardwareAddr": "Nn3aakR4",
            "Flags": 19,
            "AltAddrs": null
        },
        "awdl0": {
            "Index": 13,
            "MTU": 1500,
            "Name": "awdl0",
            "HardwareAddr": "Irenp52E",
            "Flags": 18,
            "AltAddrs": null
        },
        "bridge0": {
            "Index": 19,
            "MTU": 1500,
            "Name": "bridge0",
            "HardwareAddr": "gsdpIUgB",
            "Flags": 19,
            "AltAddrs": null
        },
        "en0": {
            "Index": 10,
            "MTU": 1500,
            "Name": "en0",
            "HardwareAddr": "FH3aakR4",
            "Flags": 19,
            "AltAddrs": null
        },
        "en1": {
            "Index": 14,
            "MTU": 1500,
            "Name": "en1",
            "HardwareAddr": "gsdpIUgB",
            "Flags": 19,
            "AltAddrs": null
        },
        "en2": {
            "Index": 15,
            "MTU": 1500,
            "Name": "en2",
            "HardwareAddr": "gsdpIUgA",
            "Flags": 19,
            "AltAddrs": null
        },
        "en3": {
            "Index": 16,
            "MTU": 1500,
            "Name": "en3",
            "HardwareAddr": "gsdpIUgF",
            "Flags": 19,
            "AltAddrs": null
        },
        "en4": {
            "Index": 17,
            "MTU": 1500,
            "Name": "en4",
            "HardwareAddr": "gsdpIUgE",
            "Flags": 19,
            "AltAddrs": null
        },
        "en5": {
            "Index": 8,
            "MTU": 1500,
            "Name": "en5",
            "HardwareAddr": "rN5IABEi",
            "Flags": 19,
            "AltAddrs": null
        },
        "en8": {
            "Index": 18,
            "MTU": 1500,
            "Name": "en8",
            "HardwareAddr": "AOCZAAb1",
            "Flags": 19,
            "AltAddrs": null
        },
        "feth1724": {
            "Index": 12,
            "MTU": 2800,
            "Name": "feth1724",
            "HardwareAddr": "+i1BU4v0",
            "Flags": 19,
            "AltAddrs": null
        },
        "feth6724": {
            "Index": 11,
            "MTU": 2800,
            "Name": "feth6724",
            "HardwareAddr": "ZmV0aBpE",
            "Flags": 19,
            "AltAddrs": null
        },
        "gif0": {
            "Index": 2,
            "MTU": 1280,
            "Name": "gif0",
            "HardwareAddr": null,
            "Flags": 24,
            "AltAddrs": null
        },
        "llw0": {
            "Index": 27,
            "MTU": 1500,
            "Name": "llw0",
            "HardwareAddr": "Irenp52E",
            "Flags": 19,
            "AltAddrs": null
        },
        "lo0": {
            "Index": 1,
            "MTU": 16384,
            "Name": "lo0",
            "HardwareAddr": null,
            "Flags": 21,
            "AltAddrs": null
        },
        "ppp0": {
            "Index": 25,
            "MTU": 1492,
            "Name": "ppp0",
            "HardwareAddr": null,
            "Flags": 25,
            "AltAddrs": null
        },
        "stf0": {
            "Index": 3,
            "MTU": 1280,
            "Name": "stf0",
            "HardwareAddr": null,
            "Flags": 0,
            "AltAddrs": null
        },
        "utun0": {
            "Index": 20,
            "MTU": 1380,
            "Name": "utun0",
            "HardwareAddr": null,
            "Flags": 25,
            "AltAddrs": null
        },
        "utun1": {
            "Index": 21,
            "MTU": 2000,
            "Name": "utun1",
            "HardwareAddr": null,
            "Flags": 25,
            "AltAddrs": null
        },
        "utun2": {
            "Index": 22,
            "MTU": 1000,
            "Name": "utun2",
            "HardwareAddr": null,
            "Flags": 25,
            "AltAddrs": null
        },
        "vnic0": {
            "Index": 23,
            "MTU": 1500,
            "Name": "vnic0",
            "HardwareAddr": "ABxCAAAI",
            "Flags": 19,
            "AltAddrs": null
        },
        "vnic1": {
            "Index": 24,
            "MTU": 1500,
            "Name": "vnic1",
            "HardwareAddr": "ABxCAAAJ",
            "Flags": 19,
            "AltAddrs": null
        }
    },
    "HaveV6": true,
    "HaveV4": true,
    "IsExpensive": false,
    "DefaultRouteInterface": "feth1724",
    "HTTPProxy": "",
    "PAC": ""
}
```

hope GitHub collapses this...

I believe something's mishandled in `net/interfaces/interfaces_darwin.go`. I was quite sleepy and went to bed though... didn't dig deep enough into it.

**Comment by frankli0324** [comment]
any updates please?

**Comment by DentonGentry** [comment]
To set expectations: the additional details were added on Saturday, literally Christmas Day. We are not fully staffed during the holiday period and may not have updates right away.

**Comment by bradfitz** [comment]
> LogID: [redacted]

Can you not redact that? That would help.

**Comment by frankli0324** [comment]
345751b760c01f38985bdf94947e47314e74f7a42578a8027249b4bb8baf3b6a

**Comment by DentonGentry** [comment]
"P3 Can't Get Started" is the severity: Tailscale cannot start because it cannot connect to control.

**Comment by frankli0324** [comment]
oh.. I thought it was saying can't get started fixing the issue(

**Comment by DentonGentry** [comment]
I believe this was fixed in Tailscale 1.32.1 and later by https://github.com/tailscale/tailscale/pull/5940
